docs/res/guides/custom_model_template.rst

```diff
@@ -3,35 +3,33 @@
 Advanced : Declaring a custom model
 =============================================
 
-To add a custom model, you will need a **.py** file with the following structure to be placed in the *napari_cellseg3d/models* folder:
+.. warning::
+    **WIP** : Adding new models is still a work in progress and will likely not work simply by adding the model in the plugin.
 
-.. note::
-    **WIP** : Currently you must modify :ref:`model_framework.py` as well : import your model class and add it to the ``model_dict`` attribute
-
-::
-
-    def get_net():
-        return ModelClass # should return the class of the model,
-        # for example SegResNet or UNET
+    Please `file an issue`_ if you would like to add a custom model and we will help you get it working.
 
+To add a custom model, you will need a **.py** file with the following structure to be placed in the *napari_cellseg3d/models* folder::
 
-    def get_weights_file():
-        return "weights_file.pth" # name of the weights file for the model,
-        # which should be in *napari_cellseg3d/models/pretrained*
+    class ModelTemplate_(ABC): # replace ABC with your PyTorch model class name
+        use_default_training = True  # not needed for now, will serve for WNet training if added to the plugin
+        weights_file = (
+            "model_template.pth"  # specify the file name of the weights file only
+        ) # download URL goes in pretrained_models.json
 
+        @abstractmethod
+        def __init__(
+            self, input_image_size, in_channels=1, out_channels=1, **kwargs
+        ):
+            """Reimplement this as needed; only include input_image_size if necessary. For now only in/out channels = 1 is supported."""
+            pass
 
-    def get_output(model, input):
-        out = model(input) # should return the model's output as [C, N, D,H,W]
-        # (C: channel, N, batch size, D,H,W : depth, height, width)
-        return out
+        @abstractmethod
+        def forward(self, x):
+            """Reimplement this as needed. Ensure that output is a torch tensor with dims (batch, channels, z, y, x)."""
+            pass
 
 
-    def get_validation(model, val_inputs):
-        val_outputs = model(val_inputs) # should return the proper type for validation
-        # with sliding_window_inference from MONAI
-        return val_outputs
-
+.. note::
+    **WIP** : Currently you must modify :ref:`model_framework.py` as well : import your model class and add it to the ``model_dict`` attribute
```

Comment:
Whats the reason for also changing model_framework?

Reply:
Basically, it's needed so it shows up in the menus. IIRC adding it to config (which would be more centralized and arguably better UX) leads to circular import issues so I moved it there

Reply:
@CYHSM We could auto-"scan" for classes in the models folder though, but it's not super hard to edit rn. And if you're adding models you'd know how to import your class and add it to the dict I guess